MobileWallet/TariLib/Tor/OnionManager.swift

```diff
@@ -170,7 +170,7 @@ public class OnionManager: NSObject {
             self.torController = TorController(socketHost: OnionManager.CONTROL_ADDRESS, port: OnionManager.CONTROL_PORT)
         }
 
-        if ((self.torThread == nil) || (self.torThread?.isCancelled ?? true)) {
+        if (self.torThread == nil || self.torThread?.isCancelled == true) && TorThread.active?.isExecuting != true {
```

Comment:
What does this achieve? Or what is it fixing?

Reply:
when we call thread.cancel() we set only flag but thread still going. This expression checks if thread has tasks.

Reply:
Ok cool, were you able to replicate the issue and then test that this fixes it every time?

Reply:
@Jasonvdb  When I deleted this code, I managed to play this crash. Then I brought it back and the crash was never reproduced again.